internal/controller/controller.go

```diff
@@ -68,18 +68,24 @@ func (c *Controller) Start(ctx context.Context) error {
 		return fmt.Errorf("failed to parse the included tracers: %w", err)
 	}
 
-	metadataCollector := hostmetadata.NewCollector(c.config.CollAgentAddr)
-	metadataCollector.AddCustomData("os.type", "linux")
+	var metadataCollector *hostmetadata.Collector
```

Comment:
The host metadata collector shouldn't be needed anymore. I will be removing it in another PR (not here, to keep things focused around a single thing).